src/api/client/Assets.ts

```diff
@@ -250,4 +257,53 @@ export class Assets {
       next,
     };
   }
+
+  /**
+   * Retrieve all the Asset Global Metadata on chain.
+   * This includes metadata id, name and optional specs
+   */
+  public async getGlobalMetadataKeys(): Promise<GlobalMetadataKey[]> {
+    const {
+      context: {
+        polymeshApi: {
+          query: {
+            asset: { assetMetadataGlobalKeyToName, assetMetadataGlobalSpecs },
+          },
+        },
+      },
+    } = this;
+
+    const [keyToNameEntries, specsEntries] = await Promise.all([
```

Comment:
I tried the queryMulti but it was getting tougher to make it type safe since the query fetch all entries instead of a specific value. I too saw that comment from Jere suggesting queryMulti after which I changed it at one of the places but wasn't able to replace this. Maybe later I'll give it another try